pas de message claire si le token n'est pas défini, seulement l'erreur python

11/12/2022 at 15:43:35 | [ERROR] 🤕 Uh, there is an unexpected error somewhere: expected token to be a str, received <class 'NoneType'> instead (<class 'type'>)

--- a/core/config.py
+++ b/core/config.py
@@ -5,11 +5,11 @@ utiliser, modifier et/ou redistribuer ce programme sous les conditions
 de la licence CeCILL diffusée sur le site "http://www.cecill.info".
 """
 
-import os
 import importlib
+import os
+import sys
 
 import yaml
-
 from LRFutils import color
 
 from core import setup_logger
@@ -21,13 +21,20 @@ _global_config = {}
 
 logger = setup_logger('config')
 
-# Check basic requirements and start this script if something is missing
 def check():
+    "Check basic requirements and start the setup script if something is missing"
     if not os.path.isfile("config.yaml"):
         print(" ")
         logger.warning("⛔ The bot is not correctly setup. Running setup script...")
         os.system("python3 setup.py")
-        exit()
+        sys.exit()
+    try:
+        get("bot.token")
+    except KeyError:
+        print(" ")
+        logger.warning("⛔ The bot token is missing. Running setup script...")
+        os.system("python3 setup.py")
+        sys.exit()
 
 
 def get(config: str):
@@ -108,7 +115,7 @@ def token_set(force_set=False):
     """Check if the token is set, if not, ask for it. Return True if the token is set,
     False if not."""
 
-    if _global_config["bot"]["token"] is not None and not force_set:
+    if _global_config["bot"].get("token") is not None and not force_set:
         choice = input(
             f"\n🔑 {color.fg.blue}A token is already set."\
                 f"Do you want to edit it? [y/N]:{color.stop} "

--- a/plugins/rss/setup.py
+++ b/plugins/rss/setup.py
@@ -29,7 +29,7 @@ def run():
                 config["twitter"]["consumer_key"] = key
 
         if config["twitter"]["consumer_key"] is not None:
-            choice = input(f"\n{blue}A consumer kkey is already set."\
+            choice = input(f"\n{blue}A Twitter consumer key is already set. "\
                 f"Do you want to edit it? [y/N]:{stop} ")
             if choice in accept:
                 set_consumer_key()
@@ -45,7 +45,7 @@ def run():
                 config["twitter"]["consumer_secret"] = secret
 
         if config["twitter"]["consumer_secret"] is not None:
-            choice = input(f"\n{blue}A consumer secret is already set."\
+            choice = input(f"\n{blue}A Twitter consumer secret is already set. "\
                 f"Do you want to edit it? [y/N]:{stop} ")
             if choice in accept:
                 set_consumer_secret()
@@ -61,7 +61,7 @@ def run():
                 config["twitter"]["access_token_key"] = key
 
         if config["twitter"]["access_token_key"] is not None:
-            choice = input(f"\n{blue}An access token key is already set."\
+            choice = input(f"\n{blue}A Twitter access token key is already set. "\
                 f"Do you want to edit it? [y/N]:{stop} ")
             if choice in accept:
                 set_access_token_key()
@@ -76,7 +76,7 @@ def run():
                 config["twitter"]["access_token_secret"] = secret
 
         if config["twitter"]["access_token_secret"] is not None:
-            choice = input(f"\n{blue}An access token secret is already set."\
+            choice = input(f"\n{blue}A Twitter access token secret is already set. "\
                 f"Do you want to edit it? [y/N]:{stop} ")
             if choice in accept:
                 set_access_token_secret()

